File: Examples/CSharp/Tags/AddImageNodeWithTag.cs
using System;

namespace Aspose.Note.Examples.CSharp.Tags
{
    public class AddImageNodeWithTag
    {
        public static void Run()
        {
            // ExStart:AddImageNodeWithTag
            // ExFor:Image
            // ExFor:Image.Tags
            // ExFor:NoteTag
            // ExFor:Page
            // ExFor:Outline
            // ExFor:OutlineElement
            // ExSummary:Shows how to add new image with tag.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tags();

            // Create an object of the Document class
            Document doc = new Document();

            // Initialize Page class object
            Page page = new Page();

            // Initialize Outline class object
            Outline outline = new Outline();

            // Initialize OutlineElement class object
            OutlineElement outlineElem = new OutlineElement();

            // Load an image
            Image image = new Image(dataDir + "icon.jpg");

            // Insert image in the document node
            outlineElem.AppendChildLast(image);
            image.Tags.Add(NoteTag.CreateYellowStar());

            // Add outline element node
            outline.AppendChildLast(outlineElem);

            // Add outline node
            page.AppendChildLast(outline);

            // Add page node
            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "AddImageNodeWithTag_out.one";
            doc.Save(dataDir);

            // ExEnd:AddImageNodeWithTag

            Console.WriteLine("\nImage node with tag added successfully.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/VisualBasic/Tags/AddImageNodeWithTag.vb
Imports System.Drawing
Imports System.IO
Imports AsposeNote = Aspose.Note
Imports System.Collections.Generic

Namespace Tags
    Public Class AddImageNodeWithTag
        Public Shared Sub Run()
            ' ExStart:AddImageNodeWithTag
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tags()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New AsposeNote.Page(doc)
            ' Initialize Outline class object
            Dim outline As New Outline(doc)
            ' Initialize OutlineElement class object
            Dim outlineElem As New OutlineElement(doc)
            ' Load an image
            Dim image As New AsposeNote.Image(doc, dataDir & Convert.ToString("icon.jpg"))
            ' Insert image in the document node
            outlineElem.AppendChild(image)
            image.Tags.Add(New NoteTag() With {
                 .Icon = TagIcon.YellowStar
            })
            ' Add outline element node
            outline.AppendChild(outlineElem)
            ' Add outline node
            page.AppendChild(outline)
            ' Add page node
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("AddImageNodeWithTag_out_.one")
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:AddImageNodeWithTag
            Console.WriteLine(Convert.ToString(vbLf & "Image node with tag added successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace